src/main/java/io/github/foundationgames/jsonem/serialization/JsonEMCodecs.java

```diff
@@ -33,66 +42,72 @@ public class JsonEMCodecs {
+                                uvSize.getY());
+        }
 
-    private static final Vector2f DEFAULT_UV_SCALE = new Vector2fComparable(1.0f, 1.0f);
+        private static final Vector2f DEFAULT_UV_SCALE = new Vector2fComparable(1.0f, 1.0f);
 
     public static final Codec<ModelCuboidData> MODEL_CUBOID_DATA = RecordCodecBuilder.create((instance) ->
             instance.group(
                     Codec.STRING.optionalFieldOf("name").forGetter(obj -> Optional.ofNullable(((ModelCuboidDataAccess) (Object) obj).jsonem$name())),
-                    Vec3f.CODEC.fieldOf("offset").forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$offset()),
-                    Vec3f.CODEC.fieldOf("dimensions").forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$dimensions()),
+                    VECTOR3F_CODEC.fieldOf("offset").forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$offset()),
+                    VECTOR3F_CODEC.fieldOf("dimensions").forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$dimensions()),
                     DILATION.optionalFieldOf("dilation", Dilation.NONE).forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$dilation()),
                     Codec.BOOL.optionalFieldOf("mirror", false).forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$mirror()),
                     VECTOR2F.fieldOf("uv").forGetter(obj -> ((ModelCuboidDataAccess)(Object)obj).jsonem$uv()),
                     VECTOR2F.optionalFieldOf("uv_scale", DEFAULT_UV_SCALE).forGetter(obj -> Vector2fComparable.of(((ModelCuboidDataAccess)(Object)obj).jsonem$uvScale()))
             ).apply(instance, JsonEMCodecs::createCuboidData)
     );
 
-    private static Codec<ModelPartData> createPartDataCodec() {
-        return RecordCodecBuilder.create((instance) ->
-                instance.group(
-                        MODEL_TRANSFORM.optionalFieldOf("transform", ModelTransform.NONE).forGetter(obj -> ((ModelPartDataAccess) obj).jsonem$transform()),
-                        Codec.list(MODEL_CUBOID_DATA).fieldOf("cuboids").forGetter(obj -> ((ModelPartDataAccess) obj).jsonem$cuboids()),
-                        LazyTypeUnboundedMapCodec.of(Codec.STRING, JsonEMCodecs::createPartDataCodec).optionalFieldOf("children", new HashMap<>()).forGetter(obj -> ((ModelPartDataAccess) obj).jsonem$children())
-                ).apply(instance, (transform, cuboids, children) -> {
-                    var data = ModelPartDataAccess.create(cuboids, transform);
-                    ((ModelPartDataAccess) data).jsonem$children().putAll(children);
-                    return data;
-                })
-        );
-    }
+        private static Codec<ModelPartData> createPartDataCodec() {
+                return RecordCodecBuilder.create((instance) -> instance.group(
+                                MODEL_TRANSFORM.optionalFieldOf("transform", ModelTransform.NONE)
```

Comment:
- Incorrect indentation
- JsonEM has no codestyle, but this reformatting seems to worsen readability even if it reduces the length of each line (reverting this would be preferred, but is unnecessary

Reply:
ah, my bad. whatever shipped with vscode's java extension did an awful job with the formatting. I'll revert to the original indentation